lang/csharp/src/apache/main/Specific/ObjectCreator.cs

```diff
@@ -90,6 +95,14 @@ private Type FindType(string name)
                     return GenericIListType.MakeGenericType(FindType(itemTypeName));
                 }
 
+                if (TryGetIDictionaryItemTypeName(name, out var itemTypesName))
+                {
+                    var key = itemTypesName[0].GetType().Name;
```

Comment:
The logic on the line 135 (if (name == t.Name || name == t.FullName ..) compare in case sensetive way. So it's not able to handle string (system name is String)

.GetType().Name helps to handle string type correctly (changing string to String).

That is **not** a flexible solution and it requires key to always be a string. Changing the type names comparing logic also does not sound like a good idea.